rpc/src/p2p.rs

```diff
@@ -1,17 +1,207 @@
 use jsonrpsee::proc_macros::rpc;
-use serde::{Deserialize, Serialize};
+use libp2p::Multiaddr;
+use serde::{de, de::Error, ser, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::str::FromStr;
+use thiserror::Error;
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct AddrInfo {
     #[serde(rename = "ID")]
-    pub id: String,
-    // TODO: multiaddr
+    pub id: PeerId,
     #[serde(rename = "Addrs")]
-    pub addrs: Vec<String>,
+    pub addrs: Vec<Multiaddr>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Stat {
+    #[serde(rename = "NumStreamsInbound")]
+    pub streams_inbound: u32,
+    #[serde(rename = "NumStreamsOutbound")]
+    pub streams_outbound: u32,
+    #[serde(rename = "NumConnsInbound")]
+    pub connections_inbound: u32,
+    #[serde(rename = "NumConnsOutbound")]
+    pub connections_outbound: u32,
+    #[serde(rename = "NumFD")]
+    pub file_descriptors: u32,
+    #[serde(rename = "Memory")]
+    pub memory: u32,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ResourceManagerStats {
+    #[serde(rename = "System")]
+    pub system: Stat,
+    #[serde(rename = "Transient")]
+    pub transient: Stat,
+    #[serde(rename = "Services")]
+    pub services: HashMap<String, Stat>,
+    #[serde(rename = "Protocols")]
+    pub protocols: HashMap<String, Stat>,
+    #[serde(rename = "Peers")]
+    pub peers: HashMap<String, Stat>,
+}
+
+#[derive(Debug, Error)]
+pub enum PeerIdError {
+    #[error("unable to decode base58 string")]
+    Base58Error,
+    #[error("libp2p error")]
+    Libp2pError(#[from] libp2p_identity::ParseError),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct PeerId(pub libp2p::PeerId);
+
+impl<'de> Deserialize<'de> for PeerId {
+    fn deserialize<D>(deserializer: D) -> Result<PeerId, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let s: String = Deserialize::deserialize(deserializer)?;
+        s.parse().map_err(D::Error::custom)
+    }
+}
+
+impl Serialize for PeerId {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        s.serialize_str(&self.0.to_base58())
+    }
+}
+
+impl FromStr for PeerId {
+    type Err = PeerIdError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = bs58::decode(s)
+            .into_vec()
+            .map_err(|_| PeerIdError::Base58Error)?;
+
+        Ok(Self(libp2p::PeerId::from_bytes(&bytes)?))
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BandwidthStats {
+    #[serde(rename = "TotalIn")]
+    pub total_in: f32,
+
+    #[serde(rename = "TotalOut")]
+    pub total_out: f32,
+
+    #[serde(rename = "RateIn")]
+    pub rate_in: f32,
+
+    #[serde(rename = "RateOut")]
+    pub rate_out: f32,
+}
```

Comment:
use `rename_all`

Reply:
done, note that these struct have been moved to [`types`](https://github.com/eigerco/celestia-node-rs/pull/52/files#diff-ed39cfb7fdc8b39ea00751ee4b418a96975ad12b235bc308b96be3246a2c8686).